vcpp-client-reference/CMakePresets.json

```diff
@@ -0,0 +1,40 @@
+﻿{
+  "version": 3,
+  "configurePresets": [
+    {
+      "name": "windows-base",
+      "description": "Target Windows with the Visual Studio development environment.",
+      "hidden": true,
+      "generator": "Ninja",
```

Comment:
we talked about this: Visual studio generated this. Do we actually need it?

Ninja is a build file system that doesnt include conditionals, making it actually simpler than makefiles. in this way its a perfect artifact for a build system generator, as it is the absolute most rudimentary scripting language. Your complexity gets pushed into cmake time, and your build scripts (ninja files) are as heavily platform-specific and target-specific as possible.

Reply:
- Without this `CMakePresets.json`, Visual Studio's defaults will still work. 
- Not sure if I would call Ninja a scripting language. But yes, all the decision-making bits are pushed into a higher-level build system such as CMake. Ninja itself can remain simple and be optimized for DAG computation and parallelizing the build process as much as possible.